### PR TITLE
Global metrics only switch

### PR DESCRIPTION
**Pull request by fenek**
This PR adds `all_metrics_are_global` switch.

Proposed changes include:
* `mongoose_metrics` got some serious refactoring.
* New option, when enabled, merges all per-host metrics into global equivalents, effectively reducing CPU and memory footprint in clusters with lots of XMPP domains supported.
* A side effect is an unification of metrics names. Host part (or `global` atom) always comes first now.
* Only unit tests and metrics_api suite are updated, because remaining metrics test cases rely heavily on RPC, which doesn't offer nice and consistent way of making calls to more than one node. The new switch is enabled on dev node 2 to avoid restarts or introducing yet another test configuration
* Updated documentation, including a note about performance problem the new switch solves.

Not all metrics test suites are modified to verify both nodes (with switch enabled and disabled) because they rely on many RPC calls and this mechanism is not ready yet for flexible calls to any node.

### CODE REVIEW

File: test/ejabberd_tests/tests/metrics_helper.erl [hunk 1]
```diff
@@ -19,12 +17,57 @@ get_counter_value({_, _} = Counter) ->
...
+make_global_group_name(GN) ->
+    list_to_atom(atom_to_list(GN) ++ "_all_metrics_are_global").
+
+make_global_groups_names(GroupsNames) ->
+    [{group, make_global_group_name(GN)} || GN <- GroupsNames].
+
+make_global_groups(Groups) ->
+    [{make_global_group_name(GN), Opts, Cases} || {GN, Opts, Cases} <- Groups].
+
+userspec(User1Count, Config) ->
```
Comment (michalwski) [inline test/ejabberd_tests/tests/metrics_helper.erl]: This function should accept the same user specs as `escalus:story/3` to make things simpler.
And BTW, why exactly do we need this function?

Comment (fenek) [inline test/ejabberd_tests/tests/metrics_helper.erl]: The purpose of this function is to return `escalus:story/3`-compatible userspec, based on presence (or absence) of `all_metrics_are_global` flag in test config. When it's present, users from node 2 are used in spec. This makes this choice transparent for actual suite.

Comment (michalwski) [inline test/ejabberd_tests/tests/metrics_helper.erl]: Ok, I see it now, it's not the easiest thing to follow. From what I understand it assumes that the `escalus_users` in `Config` contains only correct users. Probably a word of explanation will make things easier for others.

Comment (fenek) [inline test/ejabberd_tests/tests/metrics_helper.erl]: Word of explanation like explaining it in a comment to this function, right? No problem.

Comment (michalwski) [inline test/ejabberd_tests/tests/metrics_helper.erl]: Yes, that's what I meant :)

**Pull request merged**